docs/billing.md

```diff
@@ -0,0 +1,54 @@
+#Billing guide
+Cruddl can be configured to log the creation of business objects for specific type.
+This can be used for applications, where a customer is billed based on the number of created business objects.
+
+## Configuration
+
+A cruddl project contains GraphQL schema files and metadata (json/yaml) files. BillingEntities are defined in metadata files within the `billing` object as follows:
+
+```json
+{
+    "billing": {
+        "billingEntities": [
+            {
+                "typeName": "Order",
+                "keyFieldName": "orderNumber"
+            }
+        ]
+    }
+}
+```
+
+A BillingEntity is defined with a `typeName` and a `keyFieldName`. If no `keyFieldName` is supplied, the `@key` field of the type is used.
+
+Only RootEntities can be defined as BillingEntities.
+
+## How it works
+
+Whenever a create mutation is called for a rootEntity that is configured as BillingEntity, a new entry in the `billingEntities` Database-Collection is created.
+
+An entry is also created, if an update mutation is called for a rootEntity that is configured as BillingEntity, where the keyField is changed.
+
+In both cases, if there already is an entry with the same keyField value, no billing entry is created.
+
+##Logged Entries
+
+A billing entry has the following structure:
+
+```json
+{
+    "keyField": "000006",
```

Comment:
also, I'd call this `key` because it's not the field, it's the actual key. `keyField` sounds like it holds the name of the key field (e.g. `deliveryNumber`)